test/unit/ConstantProductPair.t.sol

```diff
@@ -547,8 +549,62 @@ contract ConstantProductPairTest is BaseTest
         uint256 lTotalSupply = _constantProductPair.totalSupply();
         assertEq(lTotalSupply, type(uint112).max);
 
-        (, int112 lAccLiq1, ) = _constantProductPair.observations(0);
-        (, int112 lAccLiq2, ) = _constantProductPair.observations(_constantProductPair.index());
+        (, , int112 lAccLiq1, ) = _constantProductPair.observations(0);
+        (, , int112 lAccLiq2, ) = _constantProductPair.observations(_constantProductPair.index());
         assertApproxEqRel(type(uint112).max, LogCompression.fromLowResLog( (lAccLiq2 - lAccLiq1) / 5), 0.0001e18);
     }
+
+    function testOracle_ClampedPrice() public
+    {
+        // arrange
+        _stepTime(5);
+        uint256 lSwapAmt = 100e18;
+        _tokenB.mint(address(_constantProductPair), lSwapAmt);
+        _constantProductPair.swap(-int256(lSwapAmt), true, address(this), bytes(""));
+
+        // sanity
+        assertEq(_constantProductPair.prevClampedPrice(), 1e18);
+
+        // act
+        _stepTime(5);
+        _constantProductPair.sync();
+
+        // sanity
+        assertEq(_constantProductPair.prevClampedPrice(), 1.0025e18);
+
+        // assert
+        (int112 lAccRawLogPrice0, int56 lAccClampedLogPrice0, , uint32 lTimestamp0) = _constantProductPair.observations(0);
+        (int112 lAccRawLogPrice1, int56 lAccClampedLogPrice1, , uint32 lTimestamp1) = _constantProductPair.observations(1);
+
+        // act
+        _stepTime(5);
+        _constantProductPair.sync();
+
+        // assert - make sure that the prevClampedPrice is keeping up
+        (int112 lAccRawLogPrice2, int56 lAccClampedLogPrice2, , uint32 lTimestamp2) = _constantProductPair.observations(2);
+        assertEq(_constantProductPair.prevClampedPrice(), 1.00500625e18);
```

Comment:
Id be interested in asserting the the unclamped price is more extreme. Because that would show that clamping has occurred, similar to how you wrote the next test

Reply:
okay I added an `assert` to show that the unclamped TWAP price is a lot bigger